src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -415,9 +415,124 @@ export class StudyViewPageStore {
         this._comparisonGroupsChangeCount += 1;
     }
 
-    private createPieChartComparisonSession(
+    private async createNumberAttributeComparisonSession(
         clinicalAttribute:ClinicalAttribute,
-        clinicalAttributeValues:{ value:string, color:string}[],
+        statusCallback:(phase:LoadingPhase)=>void
+    ) {
+        statusCallback(LoadingPhase.DOWNLOADING_GROUPS);
+        return new Promise<string>((resolve)=>{
+            onMobxPromise(this.selectedSamples,
+                async (selectedSamples)=>{
+                    // get clinical data for the given attribute
+                    const entityIdKey = (clinicalAttribute.patientAttribute ? "patientId" : "sampleId")
+                    const data = await client.fetchClinicalDataUsingPOST({
+                        clinicalDataType: clinicalAttribute.patientAttribute ? "PATIENT" : "SAMPLE",
+                        clinicalDataMultiStudyFilter: {
+                            attributeIds: [clinicalAttribute.clinicalAttributeId],
+                            identifiers: selectedSamples.map(s=>({ studyId: s.studyId, entityId: s[entityIdKey] }))
+                        }
+                    });
+
+                    // sort values
+                    // group into halves, thirds, or fourths, depending on how many distinct values there are
+                    const groupedByValue = _.groupBy(data, d=>parseFloat(d.value));
+                    const distinctValues = _.sortBy(Object.keys(groupedByValue).map(v=>parseFloat(v)));
+
+                    statusCallback(LoadingPhase.CREATING_SESSION);
+                    // create groups using data
+                    let groups:SessionGroupData[];
+                    let patientToSamples:{[uniquePatientKey:string]:SampleIdentifier[]} = {};
+                    if (clinicalAttribute.patientAttribute) {
+                        patientToSamples = _.groupBy(selectedSamples, s=>s.uniquePatientKey);
+                    }
+                    switch (distinctValues.length) {
+                        case 1:
+                        case 2:
+                        case 3:
```

Comment:
We have a threshold set for bar chart at backend if the unique value number is 5 then we don't group. We could use that here as well.

Reply:
since we're looking at quartiles, I think 4 makes sense, no?

Reply:
It does, but when there are only five distinct values probably makes more sense showing five groups. But again, back to original discussion which we vote down. So no need to make it complicated. What you are doing right now is correct.